### DISCUSSION TITLE
How do I get constructor parameters to indent to the same level as it's first parameter?

### DISCUSSION THREAD

**Discussion by Xevion** [eslint/eslint#15819] [Q&A] [answered]
I'm using **JetBrains Webstorm** with ESLint and I'd like my constructor parameters to all have the same indent level as it's first parameter. See below:

```ts
    constructor(description: string, status: number, errorId: string,
                extraData: Record<string, unknown> | null = null) {
        super();
```
`extraData` matches the indent level of `description`, but unfortunately, ESLint doesn't like this and wants `extraData`'s indent level to match the `super();` statement.

![image](https://user-images.githubusercontent.com/44609630/165803773-8a1392f2-0fbe-4c3e-a8b3-02e24e8918c7.png)

Here are my current rules (shortened to what's pertinent):

```ts
module.exports = {
    rules: {
        "indent": [
            "error", 4, {
                "SwitchCase": 1,
                "FunctionDeclaration": {
                    "parameters": "first",
                },
                "ObjectExpression": "first",
            }],
    }
}
```

You can see I've tried to use indent's `FunctionDeclaration` and `ObjectExpression` rules to get this working, but unfortunately, it seems it isn't happening. I'm not sure if I've got something set or typed wrong, or if `constructors` simply aren't supported (seems unlikely given how comprehensive ESLint is, usually), but I need help.

**Comment by mdjermanovic** [comment, marked as answer]
You can set `"FunctionExpression": { "parameters": "first" }`.

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50ICBcImluZGVudFwiOiBbXG4gICAgICAgICAgICBcImVycm9yXCIsIDQsIHtcbiAgICAgICAgICAgICAgICBcIlN3aXRjaENhc2VcIjogMSxcbiAgICAgICAgICAgICAgICBcIkZ1bmN0aW9uRXhwcmVzc2lvblwiOiB7XG4gICAgICAgICAgICAgICAgICAgIFwicGFyYW1ldGVyc1wiOiBcImZpcnN0XCIsXG4gICAgICAgICAgICAgICAgfVxuICAgICAgICAgICAgfV1cbiovXG5cbmNsYXNzIEMgZXh0ZW5kcyBCIHtcbiAgICBjb25zdHJ1Y3RvcihkZXNjcmlwdGlvbiwgc3RhdHVzLCBlcnJvcklkLFxuICAgICAgICAgICAgICAgIGV4dHJhRGF0YSA9IG51bGwpIHtcbiAgICAgICAgc3VwZXIoKTtcbiAgICB9XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

**Reply by Xevion** [reply]
This works. 

Side note: Is it just me, or is calling a `constructor` a "function expression" feel a bit wrong compared to calling it a "function declaration"?
Neither seems truly etymologically correct (compared to something like "constructor declaration"), but between the two, I'd place `constructor` under "function declaration" given that it is essentially a declaration (since there is a function body).
